crates/lib3h/src/gateway/gateway_dht.rs

```diff
@@ -96,12 +97,36 @@ impl P2pGateway {
                 unreachable!();
             }
             DhtRequestToParent::RequestEntry(_) => {
-                unreachable!();
+                // no-op
             }
         }
         // Forward to parent
-        self.endpoint_self
-            .publish(span, GatewayRequestToParent::Dht(payload))?;
+        self.endpoint_self.request(
+            span,
+            GatewayRequestToParent::Dht(payload),
+            Box::new(|me, response| {
+                trace!("Received requestEntry response in Gateway");
+                let dht_response = match response {
+                    GhostCallbackData::Response(Ok(GatewayRequestToParentResponse::Dht(d))) => d,
+                    GhostCallbackData::Response(Err(e)) => {
+                        panic!("Got error on GatewayRequest DHT: {:?} ", e);
+                    }
+                    GhostCallbackData::Timeout(_bt) => {
+                        // Not all requests expects a response so timeouts are expected
```

Comment:
Would it make sense to distinguish between the ones that do and those that don't here then and panic on some of them?

Reply:
I'll move the request within the RequestEntry variant and have the rest be a publish as before